scripts/tools/EXT_Exp_MetricTracer.rst

```diff
@@ -0,0 +1,167 @@
+<%
+import re
+from templates import helper as th
+%><%
+    OneApi=tags['$OneApi']
+    x=tags['$x']
+    X=x.upper()
+    t=tags['$t']
+    T=t.upper()
+%>
+:orphan:
+
+.. _ZET_experimental_metric_tracer:
+
+==========================================
+Metric Tracer Experimental Extension
+==========================================
+
+API
+----
+* Enumerations
+
+    * ${t}_metric_group_sampling_type_flags_t
+
+    New value ${T}_METRIC_GROUP_SAMPLING_TYPE_FLAG_EXP_TRACER_BASED
+
+    * ${t}_metric_type_t
+
+    New Values
+        ${T}_METRIC_TYPE_EXP_EVENT_NO_VALUE   
+            Metric type:  have only timestamp and value has no meaning.
+        ${T}_METRIC_TYPE_EXP_EVENT_START
+            Metric type: the first event of a start/stop event pair.
+        ${T}_METRIC_TYPE_EXP_EVENT_END
+            Metric type: the second event of a start/stop event pair.
+        ${T}_METRIC_TYPE_EXP_EVENT_MONOTONIC_WRAPS_VALUE
+            Metric type: value of the event is a monotonic increasing value that can wrap around.
+
+
+* Structures
+
+    * ${t}_metric_tracer_exp_desc_t
+    * ${t}_metric_entry_exp_t
+
+* Functions
+
+    * ${t}MetricTracerCreateExp
+    * ${t}MetricTracerDestroyExp
+    * ${t}MetricTracerEnableExp
+    * ${t}MetricTracerDisableExp
+    * ${t}MetricTracerReadDataExp
+    * ${t}MetricDecoderCreateExp
+    * ${t}MetricDecoderDestroyExp
+    * ${t}MetricDecoderGetDecodableMetricsExp
+    * ${t}MetricTracerDecodeExp
+
+Metric Tracer
+~~~~~~~~~~~~~~~~~~~
+
```

Comment:
it might be good to expand on what a "metric tracer" itself is in this section. from statements below it's not clear/obvious to me what Metric Tracer or Metric Decoder is and how they compliment existing metrics interfaces

Reply:
short summary it is expanding the the two existing collection methods (streamer and query) https://spec.oneapi.io/level-zero/latest/tools/PROG.html#metric-streamer  with a new paradigm that allows collecting events of asynchronous nature. I will expand the description and submit.